Index: empserver/src/lib/subs/pr.c
--- empserver/src/lib/subs/pr.c	27 May 2005 16:31:19 -0000	1.12
+++ empserver/src/lib/subs/pr.c	28 May 2005 08:32:40 -0000
@@ -57,6 +57,13 @@
 
 static void outid(struct player *pl, int n);
 
+/* Hack for nightlybuild */
+char *
+ctime(const time_t *clock)
+{
+	return "Thu Jan  1 00:00:00 GMT 1970\n";
+}
+
 /*VARARGS*/
 void
 pr(char *format, ...)
@@ -294,7 +301,7 @@
 void
 prprompt(int min, int btu)
 {
-    pr_id(player, C_PROMPT, "%d %d\n", min, btu);
+    pr_id(player, C_PROMPT, "0 640\n"); /* Hack for nightlybuild */
 }
 
 void
