Index: empserver/src/lib/subs/pr.c
--- empserver/src/lib/subs/pr.c	18 Aug 2007 17:03:13 -0000	1.41
+++ empserver/src/lib/subs/pr.c	9 Sep 2007 02:42:51 -0000
@@ -68,6 +68,13 @@
 static void upr_player(struct player *pl, int id, char *buf);
 static void outid(struct player *pl, int n);
 
+/* Hack for nightlybuild */
+char *
+ctime(const time_t *clock)
+{
+    return "Thu Jan  1 00:00:00 GMT 1970\n";
+}
+
 /*
  * Print to current player similar to printf().
  * Use printf-style FORMAT with the optional arguments.
@@ -351,7 +358,7 @@
 void
 prprompt(int min, int btu)
 {
-    pr_id(player, C_PROMPT, "%d %d\n", min, btu);
+    pr_id(player, C_PROMPT, "0 640\n"); /* Hack for nightlybuild */
 }
 
 /*
