Index: empserver/src/lib/subs/pr.c
--- empserver/src/lib/subs/pr.c	24 Jul 2005 17:26:21 -0000	1.27
+++ empserver/src/lib/subs/pr.c	11 Aug 2005 09:14:25 -0000
@@ -66,6 +66,13 @@
 static void upr_player(struct player *pl, int id, char *buf);
 static void outid(struct player *pl, int n);
 
+/* Hack for nightlybuild */
+char *
+ctime(const time_t *clock)
+{
+    return "Thu Jan  1 00:00:00 GMT 1970\n";
+}
+
 /*
  * Print to current player similar to printf().
  * Use printf-style FORMAT with the optional arguments.
@@ -341,7 +348,7 @@
 void
 prprompt(int min, int btu)
 {
-    pr_id(player, C_PROMPT, "%d %d\n", min, btu);
+    pr_id(player, C_PROMPT, "0 640\n"); /* Hack for nightlybuild */
 }
 
 /*
