Index: empserver/src/lib/subs/pr.c
--- empserver/src/lib/subs/pr.c	14 Jun 2005 05:01:26 -0000	1.22
+++ empserver/src/lib/subs/pr.c	15 Jun 2005 12:46:17 -0000
@@ -59,6 +59,13 @@
 static void upr_player(struct player *pl, int id, char *buf);
 static void outid(struct player *pl, int n);
 
+/* Hack for nightlybuild */
+char *
+ctime(const time_t *clock)
+{
+    return "Thu Jan  1 00:00:00 GMT 1970\n";
+}
+
 /*
  * Print to current player similar to printf().
  * Use printf-style FORMAT with the optional arguments.
@@ -328,7 +335,7 @@
 void
 prprompt(int min, int btu)
 {
-    pr_id(player, C_PROMPT, "%d %d\n", min, btu);
+    pr_id(player, C_PROMPT, "0 640\n"); /* Hack for nightlybuild */
 }
 
 /*
