Index: empserver/Make.sysdefs
--- empserver/Make.sysdefs	Sat Mar  6 13:50:20 2004
+++ empserver/Make.sysdefs	Sat Mar  6 13:50:02 2004
@@ -96,7 +96,7 @@
 RTPCLFLAGS = 
 RTPCMASTER = GLOBALCFLAGS="$(RTPCCFLAGS)" GLOBALLFLAGS="$(RTPCLFLAGS)"
 
-SOLCFLAGS = -g -DSVR4 -DRel4 -DUCONTEXT -Dsolaris
+SOLCFLAGS = -g -DSVR4 -DRel4 -DUCONTEXT -Dsolaris $(GCCWARNFLAGS) -Wno-char-subscripts
 SOLLFLAGS = -lnsl -lsocket -lthread
 SOLMASTER = GLOBALCFLAGS="$(SOLCFLAGS)" GLOBALLFLAGS="$(SOLLFLAGS)"
 
