diff --git a/src/lib/global/path.c.in b/src/lib/global/path.c.in
index a761735..77b9c50 100644
--- a/src/lib/global/path.c.in
+++ b/src/lib/global/path.c.in
@@ -38,7 +38,7 @@
 #include "optlist.h"
 
 /* econfig file to try when user doesn't specify one */
-char dflt_econfig[] = "@econfig@";
+char dflt_econfig[] = "c:/MinGW@econfig@";
 
 /*
  * Where to find configuration (absolute file name)
@@ -53,17 +53,17 @@ char *custom_tables = "";
 char *schedulefil;
 
 /* Where to find built-in configuration tables (relative to configdir) */
-char *builtindir_conf = "@builtindir@";
+char *builtindir_conf = "c:/MinGW@builtindir@";
 /* Where to find built-in configuration tables (absolute) */
 char *builtindir;
 
 /* Where to find info pages (relative to configdir) */
-char *infodir_conf = "@einfodir@";
+char *infodir_conf = "c:/MinGW@einfodir@";
 /* Where to find info pages (absolute) */
 char *infodir;
 
 /* Where this game's data is stored (relative to configdir) */
-char *gamedir_conf = "@gamedir@";
+char *gamedir_conf = "c:/MinGW@gamedir@";
 /* Where this game's data is stored (absolute) */
 char *gamedir;
 
